### PR TITLE
refactor: clean up vexide-startup

### PR DESCRIPTION
**Pull request by Tropix126**
## Describe the changes this PR makes. Why should it be merged?
- Moves code signature stuff into its own module. This is then re-exported from lib.rs.
- Removed `zero_bss`, replacing it with a slice fill to simplify stuff.
- Changed some comments around to be more clear.

Needs hardware testing, specifically around the changes to the `bss` zero process.

### CODE REVIEW

**Comment by Tropix126** [comment]
So since the BSS code is now nonvolatile, it's being optimzied into a `memset` in release mode, but I don't think that's a problem?

**Comment by Tropix126** [comment]
TODO: Prefix some `__overwriter_aeabi_memcpy` symbols to prevent clashes with the ARM memcpy compiler-rt intrinsic.

**Comment by GamingLiamStudios** [comment]
> Prefix some `__overwriter_aeabi_memcpy` symbols to prevent clashes

From my (minimal) testing, the 4 and 8 byte specialisations either don't need to exist, or can be de-globalised, as it seems they aren't referenced anywhere.

**Comment by Tropix126** [comment]
Made some adjustments to the `#[vexide::main]` macro expansion, so i'll need to update tests.

File: packages/vexide-startup/src/patcher/mod.rs [hunk 1]
```diff
@@ -97,113 +108,116 @@ pub(crate) unsafe fn patch() {
...
+}
 
-        while !new.is_empty() {
-            let processed = match state {
-                PatcherState::Initial => {
-                    state = PatcherState::Add(patch.read_varint().unwrap());
-                    0
+/// Bipatch Algorithm
+///
+/// This function writes into `new` given an older `base` reference buffer and a [bidiff]-compatible
```
Comment (Gavin-Niederman) [inline packages/vexide-startup/src/patcher/mod.rs:163]: probably borked bidiff link

**Review by Gavin-Niederman — CHANGES_REQUESTED** [review]
very lgtm if tested

**Review by Gavin-Niederman — APPROVED** [review]
LGTM
